docs/docs/create-source-plugin.md

```diff
@@ -100,3 +100,56 @@ package provides a set of helper functions to generate Node objects with the
 required fields. This includes automatically generating fields like node IDs
 and the `contentDigest` MD5 hash, keeping your code focused on data gathering,
 not boilerplate.
+
+## Advanced
+
+### Adding relationships between nodes
+
+Gatsby source plugins not only create nodes, they also create relationships between nodes that are exposed to GraphQL queries.
+
+There are two ways of adding node relationships in Gatsby: (1) hierarchical (parent-child) or (2) foreign-key based. 
+
+#### Hierarchical relationships
+
+An example of a hierarchical relationship would be a `transformer` plugin changing a file node's markdown string into a `childMarkdownRemark` node that is attached to that fileNode. The markdown node doesn't make sense outside of the file it belongs to. 
+
+When a parent node is deleted, Gatsby deletes all of the child nodes (and their child nodes, and so on). 
+
+*Creating the relationship*
+
+In order to create a parent/child relationship, when calling `createNode` for the child node the object that is passed in should have a `parent` key with the value set to the parent node's `id`. After this, call the `createParentChildLink` function exported inside `actions`.
+
+*Example*
+
+[Here's an example](https://github.com/gatsbyjs/gatsby/blob/1fb19f9ad16618acdac7eda33d295d8ceba7f393/packages/gatsby-transformer-sharp/src/on-node-create.js#L3-L25) from the `gatsby-transformer-sharp` source plugin.  
+
+#### Foreign-key relationships
+
+An example of a foreign-key relationship would be a Post that has an Author. 
+
+In this relationship, each object is a distinct entity that exists whether or not the other does, with independent schemas, and field(s) on each entity that reference the other entity -- in this case the Post would have an Author, and the Author might have Posts. The API of a service that allows complex object modelling, for example a CMS, will often allow users to relationships between entities and expose them through the API.
+
+When an object node is deleted, Gatsby _does not_ delete any referenced entities. 
```

Comment:
Follow with — "It's a source plugin's responsibility to clean up dangling entity references".